src/core/server/elasticsearch/cluster_client.ts

```diff
@@ -197,4 +205,19 @@ export class ClusterClient {
       throw new Error('Cluster client cannot be used after it has been closed.');
     }
   }
+
+  // supports both KibanaRequest & Legacy.Request while asScoped is used in LegacyPlatform
+  private getHeaders(
+    request?: KibanaRequest | Request
+  ): Record<string, string | string[] | undefined> {
+    if (!request) return {};
+    const authHeaders = this.getAuthHeaders(request);
+
+    const headers = filterHeaders(
+      ensureRawRequest(request).headers,
+      this.config.requestHeadersWhitelist
+    );
+
+    return { ...headers, ...authHeaders };
```

Comment:
if there are no `authHeaders` returned from `registerAuth` or `registerAuth` wasn't called, we will use headers from the incoming request.